storage/client_raft_test.go

```diff
@@ -651,3 +652,30 @@ func TestReplicateAddAndRemove(t *testing.T) {
 		verify([]int64{39, 0, 39, 39})
 	}
 }
+
+// TestRaftHeartbeats verifies that coalesced heartbeats are correctly
+// suppressing elections in an idle cluster.
+func TestRaftHeartbeats(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	mtc := startMultiTestContext(t, 3)
+	defer mtc.Stop()
+	mtc.replicateRange(1, 0, 1, 2)
+
+	// Capture the initial term and state.
+	status := mtc.stores[0].RaftStatus(1)
+	initialTerm := status.Term
+	if status.SoftState.RaftState != raft.StateLeader {
+		t.Errorf("expected node 0 to initially be leader but was %s", status.SoftState.RaftState)
+	}
+
+	// Wait for several ticks to elapse.
+	time.Sleep(5 * mtc.makeContext().RaftTickInterval)
```

Comment:
Cases like this make me wonder if we should expose a facility that exposes these sorts of events, so that our tests can precisely listen for N events (ticks in this case), rather than sleeping for an imprecise period of time.